### DISCUSSION TITLE
field_validator is not supported?

### DISCUSSION THREAD

**Discussion by ferdousbhai** [fastapi/sqlmodel#787] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel, Field, field_validator


class Market(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    question: str
    description: str
    outcomes: list[str] = Field(
        default_factory=list, description="list of possible outcomes"
    )
    outcomePrices: list[float] = Field(
        default_factory=list, description="probabilities of each outcome"
    )
    volume: float
    categories: list[str] = Field(
        default_factory=list, description="list of categories"
    )

    @field_validator("categories", mode="before")
    def extract_category_labels(cls, categories):
        if categories:
            return [category["label"] for category in categories]
        return []

    @field_validator("outcomes", "outcomePrices", mode="before")
    def parse_lists(cls, v):
        if isinstance(v, str):
            return json.loads(v)
        return v
```

### Description

I want to pre-process some data before initializing the model. Works fine with Pydantic but seems Pydantic's 'field_validator' is missing in sqlmodel.

### Operating System

Linux

### SQLModel Version

latest

### Python Version

3.11

**Comment by northshaur** [comment]
Did you find a response separately? Facing the same issue

**Reply by joachimhuet** [reply, marked as answer]
Validation is not triggered when `table=True` by choice. See #52 for alternatives. I would suggest you to use a non table class with the validation and then inherit it from your `table=True`.
Like so:
```python
class MarketBase(SQLModel):
    id: int | None = Field(default=None, primary_key=True)
    question: str
    description: str
    outcomes: list[str] = Field(
        default_factory=list, description="list of possible outcomes"
    )
    outcomePrices: list[float] = Field(
        default_factory=list, description="probabilities of each outcome"
    )
    volume: float
    categories: list[str] = Field(
        default_factory=list, description="list of categories"
    )

    @field_validator("categories", mode="before")
    def extract_category_labels(cls, categories):
        if categories:
            return [category["label"] for category in categories]
        return []

    @field_validator("outcomes", "outcomePrices", mode="before")
    def parse_lists(cls, v):
        if isinstance(v, str):
            return json.loads(v)
        return v

class Market(MarketBase, table=True): ...
```
`MarketBase` instances should now use the validation properly. And you can create a Market instance with `Market.model_validate(market_base)`
